numpy/core/src/umath/loops.c.src

```diff
@@ -2618,32 +2618,33 @@ OBJECT_@kind@(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUS
 NPY_NO_EXPORT void
 OBJECT_sign(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED(func))
 {
-#if defined(NPY_PY3K)
     PyObject *zero = PyLong_FromLong(0);
+    PyObject *one = PyLong_FromLong(1);
+    PyObject *minus_one = PyLong_FromLong(-1);
```

Comment:
Either `one` or `minus_one` is not going to be returned, and should therefore have its refcount decreased before returning, similarly to what happens with `zero`.